provider/k8s/telemetry.go

```diff
@@ -0,0 +1,137 @@
+package k8s
+
+import (
+	"context"
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	am "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	skipParams = []string{
+		"name",
+		"rack_name",
+		"cidr",
+		"key_pair_name",
+		"internet_gateway_id",
+		"syslog",
+		"tags",
+		"vpc_id",
+		"whitelist",
+	}
+	redactedParams = strings.Join(skipParams, ",")
+)
+
+func (p *Provider) RackParams() map[string]interface{} {
+	trp, err := p.Cluster.CoreV1().ConfigMaps(p.Namespace).Get(context.TODO(), "telemetry-rack-params", am.GetOptions{})
+	if err != nil {
+		fmt.Printf("could not find rack params configmap: %v", err)
+		return nil
+	}
+
+	trs, err := p.Cluster.CoreV1().ConfigMaps(p.Namespace).Get(context.TODO(), "telemetry-rack-sync", am.GetOptions{})
+	if err != nil {
+		fmt.Printf("could not find params sync configmap: %v: Creating configmap...", err)
+
+		var hadErrorOnCreate bool
+		var noParamsToSync map[string]interface{}
+		trs, hadErrorOnCreate, noParamsToSync = createNewSyncConfigMap(trp, trs, err, p)
+		if hadErrorOnCreate {
+			return noParamsToSync
+		}
```

Comment:
if you still want to use `trs` I suggest to use a pointer, so in the `createNewSyncConfigMap` you can assigned directly to it.
also, `noParamsToSync` is always returned as nil, consider removing it.